packages/components/src/internal/components/heatmap/HeatMapDisplay.tsx

```diff
@@ -13,67 +13,68 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import React from 'react';
+import React, { FC, memo, ReactNode, useMemo } from 'react';
 import { List, Map, Set } from 'immutable';
 import { OverlayTrigger, Tooltip } from 'react-bootstrap';
 
-import { HeatMapProps } from './HeatMap';
+import { HeatMapCell, HeatMapDisplayCell, HeatMapProps } from './HeatMap';
 import { naturalSort } from '../../..';
 
+const applySort = (sortFn, labels: Set<string>): Set<string> => {
+    let sorted;
+
+    if (typeof sortFn === 'function') {
+        sorted = labels.sort(sortFn);
+    }
+    else {
+        sorted = labels.sort(naturalSort);
+    }
+
+    return sorted;
+}
+
 interface HeatMapDisplayProps extends HeatMapProps {
-    data: List<Map<string, any>>;
+    data: HeatMapCell[];
     displayMeasure?: boolean;
     onCellClick?: Function;
```

Comment:
nit: All of these `Function` types should be made stronger:

```ts
type SortComparator = (a: any, b: any) => number;

interface HeatMapDisplayProps extends HeatMapProps {
    data: HeatMapCell[];
    displayMeasure?: boolean;
    onCellClick?: (cell: HeatMapCell) => void;
    onHeaderClick?: (header: string, data: HeatMapCell[], index: number) => void;
    xSort?: SortComparator;
    ySort?: SortComparator;
    yAxisColumns: Record<string, HeatMapDisplayCell>;
    yTotalCellRenderer?: (cell: HeatMapDisplayCell) => ReactNode;
}
```

Reply:
Yes, looks good